api/proto/metrics/metrics.proto

```diff
@@ -24,6 +24,42 @@ message memoryMetric {
   double availPercentage = 5;
 }
 
+// App Memory Metric
+message AppMemoryMetric {
+    // total_MB - Memory allocated to app instance by EVE. In Steady State,
+    //  this is expected to be same as in App instance config. Eve should
+    //  report what it actually allocated / reserved instead of what is configured.
+    uint32 total_MB = 1;
+
+    // used_MB - Memory Used by app from total_MB Memory
+    //  used_MB <= total_MB
+    uint32 used_MB = 2;
+
+    // avail_MB - Available memory remaining out of Total for the App.
+    //  total_MB >= used_MB + avail_MB
+    uint32 avail_MB = 3;
+}
+
+message DeviceMemoryMetric {
+  // total_mem_MB - Total Device memory as seen by EVE.
+  uint32 total_mem_MB = 1;
+
+  // allocated_mem_apps_MB - Total memory Allocated to AppInstances
+  uint32 allocated_mem_apps_MB = 2;
+
+  // allocated_eve_MB - Total Allocated Memory for Eve
+  // This is the Max available memory for EVE on the device.
+  uint32 allocated_eve_MB = 3;
```

Comment:
We can't guarantee this being a hard limit -- we should probably put it into the comment

Reply:
Done..
I guess in that case, the expectation is that Controller would flag that as an error - so that the user can take corrective action by allocating more memory to Eve?